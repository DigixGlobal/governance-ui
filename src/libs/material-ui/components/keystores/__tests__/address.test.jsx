import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Address from '../address';

jest.mock('../../../helpers/blockie');

const address = {
  address: '0x12345',
  name: 'test address',
};

describe('<Address />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <Address address={address} />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
