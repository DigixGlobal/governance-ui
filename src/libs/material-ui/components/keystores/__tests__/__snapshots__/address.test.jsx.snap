// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> renders correctly 1`] = `
<TableRow
  as="tr"
  cellAs="td"
>
  <TableCell
    as="td"
    width="1"
  >
    <Header
      as="h4"
      image={true}
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <Image
        as="img"
        shape="rounded"
        size="mini"
        ui={true}
      />
      <HeaderContent>
        test address
        <HeaderSubheader>
          <code
            style={
              Object {
                "fontSize": "0.8em",
              }
            }
          >
            0x12345
          </code>
          <EZModal
            content={
              <QrCode
                data="0x12345"
              />
            }
            header="test address"
            size="small"
            trigger={
              <Icon
                as="i"
                name="qrcode"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              />
            }
          />
        </HeaderSubheader>
      </HeaderContent>
    </Header>
  </TableCell>
  <Connect(AddressBalances)
    address={
      Object {
        "address": "0x12345",
        "name": "test address",
      }
    }
  />
</TableRow>
`;
